libminifi/include/utils/BaseHTTPClient.h

```diff
@@ -51,38 +39,30 @@ struct HTTPProxy {
   int port = 0;
 };
 
-struct HTTPUploadCallback {
-  HTTPUploadCallback() {
-    stop = false;
-    ptr = nullptr;
-    pos = 0;
-  }
-  std::mutex mutex;
-  std::atomic<bool> stop;
-  ByteInputCallback *ptr;
-  size_t pos;
+class HTTPUploadCallback : public ByteInputCallback {
+ public:
+  template<typename... Args>
+  explicit HTTPUploadCallback(Args&& ... args) : ByteInputCallback(std::forward<Args>(args)...) {}
 
   size_t getPos() {
-    std::lock_guard<std::mutex> lock(mutex);
     return pos;
   }
```

Comment:
After removing the lock is this strictly needed anymore? The position member is public and could be used as it is. Same for HTTPReadCallback

Reply:
makes sense, removed them in https://github.com/apache/nifi-minifi-cpp/pull/1383/commits/b80c1c4c6adee5dba02fe5bd355f8732769edf99